control/sysnorm.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+"""sysnorm.py
+
+Functions for computing system norms.
+
+Routine in this module:
+
+norm
+
+Created on Thu Dec 21 08:06:12 2023
+Author: Henrik Sandberg
+"""
+
+import numpy as np
+import numpy.linalg as la
+
+import control as ct
+
+__all__ = ['norm']
+
+#------------------------------------------------------------------------------
+
+def norm(system, p=2, tol=1e-6, print_warning=True):
+    """Computes norm of system.
+    
+    Parameters
+    ----------
+    system : LTI (:class:`StateSpace` or :class:`TransferFunction`)
+        System in continuous or discrete time for which the norm should be computed.
+    p : int or str
+        Type of norm to be computed. p=2 gives the H_2 norm, and p='inf' gives the L_infinity norm.
+    tol : float
+        Relative tolerance for accuracy of L_infinity norm computation. Ignored
+        unless p='inf'.
+    print_warning : bool
+        Print warning message in case norm value may be uncertain.
+    
+    Returns
+    -------
+    norm_value : float or NoneType
+        Norm value of system (float) or None if computation could not be completed.
+   
+    Notes
+    -----
+    Does not yet compute the L_infinity norm for discrete time systems with pole(s) in z=0.
+    
+    Examples
+    --------
+    >>> Gc = ct.tf([1], [1, 2, 1])
+    >>> ct.norm(Gc,2)
+    0.5000000000000001
+    >>> ct.norm(Gc,'inf',tol=1e-10)
+    1.0000000000582077
+    """
+    G = ct.ss(system)
+    A = G.A
+    B = G.B
+    C = G.C
+    D = G.D
+    
+    #
+    # H_2-norm computation
+    #
+    if p == 2:  
+        # Continuous time case
+        if G.isctime():
+            poles_real_part = G.poles().real    
+            if any(np.isclose(poles_real_part, 0.0)):
+                if print_warning:
+                    print("Warning: Poles close to, or on, the imaginary axis. Norm value may be uncertain.")
```

Comment:
Don't use `print` here and in the error cases below. For warnings use `warnings`, if it is an error, please keep raising the exception. And be more specific than just `Exception`.